src/pcre2_substitute.c

```diff
@@ -1148,7 +1151,10 @@ do
   remembered. Do the callout if there is one and we have done an actual
   replacement. */
 
-  if (!overflowed && mcontext != NULL && mcontext->substitute_callout != NULL)
+  if (mcontext == NULL || mcontext->substitute_callout == NULL)
+    {}
```

Comment:
I don't really like these

`if ((mcontext != NULL || mcontext->substitute_callout != NULL) && !overflowed)`

Reply:
I prefer nesting. Indentation makes easier to see the logical blocks.

Reply:
Great, done.